pkg/build/api/types.go

```diff
@@ -178,6 +188,26 @@ type GitHubWebHookCause struct {
 	Secret string
 }
 
+// GitLabWebHookCause has information about a GitLab webhook that triggered a
+// build.
+type GitLabWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
+
+// BitbucketWebHookCause has information about a Bitbucket webhook that triggered a
+// build.
+type BitbucketWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
```

Comment:
have we shipped protobuf yet? we have at least [one field](https://github.com/openshift/origin/issues/13419#issuecomment-287970553) we need to fix up